core/payment/src/lib.rs

```diff
@@ -73,3 +75,30 @@ impl PaymentService {
         log::info!("Payment service stopped.");
     }
 }
+
+async fn release_allocation(db: DbExecutor) {
+    if let Ok(allocations) = db
+        .as_dao::<dao::AllocationDao>()
+        .get_filtered(None, None, None, None, None)
+        .await
+    {
+        allocations.iter().for_each(|a| {
+            let allocation = a.clone();
+            let db = db.clone();
+            let deadline = allocation.timeout.clone().unwrap_or(Utc::now());
+            tokio::task::spawn_local(async move {
+                let deadline = deadline - Utc::now();
```

Comment:
for `tokio::time::delay_for()` we use `Duration`. `duration` is fetched with this block of code :

`deadline.to_std().unwrap_or(Duration::from_secs(0))` 

`to_std()` -> This function errors when duration is less than zero. As standard library implementation is limited to non-negative values.

Reply:
Entire function `release_allocation_after()` is now in `core/payment/src/api/allocations.rs`.
As you can see below I decided to rewrite code to simplify and make it reading-friendly. In case of having deadline in the past calculated timestamp is negative and deadline stays being equal to zero. So the thread is not delayed and allocation release is exectuted instantly.

```rust
            let timestamp = timeout.timestamp() - Utc::now().timestamp();
            let mut deadline = 0u64;

            if timestamp.is_positive() {
                deadline = timestamp as u64;
            }

            tokio::time::delay_for(Duration::from_secs(deadline)).await;
```